src/validation.cpp

```diff
@@ -1007,8 +1002,7 @@ bool MemPoolAccept::Finalize(ATMPArgs& args, Workspace& ws)
                 hash.ToString(),
                 FormatMoney(nModifiedFees - nConflictingFees),
                 (int)entry->GetTxSize() - (int)nConflictingSize);
-        if (args.m_replaced_transactions)
-            args.m_replaced_transactions->push_back(it->GetSharedTx());
+        args.m_replaced_transactions.emplace_back(it->GetSharedTx());
```

Comment:
abc6ff1

Note, you might have a slight performance regression due to `emplace-back` memory allocation if container capacity is too small. Before this change, this potential reallocation wasn't the default.

I think that's okay `replaced_transactions` was already existent in the most busiest code path `ProcessMessage`.

Reply:
switched it back to `push_back` to adhere to my promise of "pure refactor"

Reply:
I think switching to `push_back` still do an automatic reallocation of the allocated storage if container capacity is too short. See https://www.cplusplus.com/reference/vector/vector/emplace_back/ and https://www.cplusplus.com/reference/vector/vector/push_back/.

My point was underscoring that the allocation was previously conditional of `m_replaced_transaction` pointer being opt-in by ATMP caller. After this change it's not anymore. Again I don't think that matters that most.